File: public/dsa/searching/negative_numbers_in_sorted_array/lib/negative_numbers_in_sorted_array.py
from bisect import bisect_left
from typing import List


class Solution:
    def getNegativeNumbersCount(self, arr: List[int]) -> int:
        return bisect_left(arr, 0)


# tests-start
arr = [-5, -3, -2, 3, 4, 6, 7, 8]
sol = Solution()
print(sol.getNegativeNumbersCount(arr))
# tests-end


'''output
3
'''

File: public/dsa/searching/negative_numbers_in_sorted_array/lib/negative_numbers_in_sorted_array.cpp
#include <bits/stdc++.h>
using namespace std;

int getNegativeNumbersCount(vector<int> &arr) {
    auto lb = lower_bound(arr.begin(), arr.end(), 0);
    int lbi = lb - arr.begin();
    return lbi;
}

// tests-start
int main() {
    vector<int> arr = {-5, -3, -2, 3, 4, 6, 7, 8};
    int result = getNegativeNumbersCount(arr);
    cout << result << endl;
    return 0;
}
// tests-end

/*output
3
*/